src/test/java/core/basesyntax/service/ShopServiceImplTest.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.service;
+
+import static core.basesyntax.storage.Storage.storageOfFruits;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.action.ActionHandler;
+import core.basesyntax.service.action.BalanceAction;
+import core.basesyntax.service.action.PurchaseAction;
+import core.basesyntax.service.action.ReturnAction;
+import core.basesyntax.service.action.SupplyAction;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.junit.jupiter.api.Test;
+
+class ShopServiceImplTest {
```

Comment:
add more tests, like
1 - empty transaction list case
2 - invalid transactions
3 - more fruit types
If you have multiple tests, don't forget to clear storage after each, in all test classes

Reply:
not fixed

Reply:
why? I added more tests

Reply:
Not fixed, basically in this class only one test

Reply:
Not fixed. See @okuzan comment. If you have any questions, you can ask them in the chat